### PR TITLE
Add sb10yd

### PR DESCRIPTION
**Pull request by KybernetikJo**
This Draft/PR adds the SLICOT Function SB10YD.f. 
A merge would close the issue #153.

It's working, following thing could be improved::

- [x] Slightly improve docstring.
- [x] Slightly improve unittests.
- [x] The order of the procedures must be restored.

### CODE REVIEW

File: slycot/tests/test_sb10yd.py [hunk 1]
```diff
@@ -3,9 +3,11 @@
 import numpy as np
 from scipy import signal
 
+import matplotlib.pyplot as plt
+
```
Comment (bnavigator) [inline slycot/tests/test_sb10yd.py:7]: Please remove this

File: slycot/tests/test_sb10yd.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import unittest
+from slycot import synthesis
+import numpy as np
+from scipy import signal
+
+class test_sb10yd(unittest.TestCase):
```
Comment (bnavigator) [inline slycot/tests/test_sb10yd.py:6]: We use pytest. No need to wrap this into a `unittest.TestCase` class.

**Comment by KybernetikJo** [comment]
@bnavigator

How I get your force-push into my local and origin branch?
I think I've done double work here.

**Comment by bnavigator** [comment]
It's alright, you did good. My force push was just a revert to (your) ddb8c03 because my attempt to resolve the merge conflict in 86032c8 had a mistake.

If there is a relevant change to your origin, you must merge or rebase your origin into your local before proceeding to push to origin.

File: slycot/synthesis.py [hunk 3]
```diff
@@ -1700,6 +1699,119 @@ def sb10jd(n,m,np,A,B,C,D,E,ldwork=None):
...
+                'TOL', 'IWORK' + hidden, 'DWORK' + hidden, 'LDWORK',
+                'ZWORK' + hidden, 'LZWORK', 'INFO' + hidden]
+
+    if ldwork is None:
+        lw1 = 2*lendat + 4*2048
+        lw2 =   lendat + 6*2048
+        mn  = min(2*lendat,2*n+1)
+        if n > 0:
+            lw3 = 2*lendat*(2*n+1) + max(2*lendat,2*n+1) + max(mn+6*n+4,2*mn+1)
+        elif n == 0:
```
Comment (bnavigator) [inline slycot/synthesis.py:1794]: Could you please cover this case in the unit tests?

Comment (KybernetikJo) [reply slycot/synthesis.py:1794]: I have to look into that.

File: slycot/tests/test_sb10yd.py (hunk 2 above)
Comment (bnavigator) [reply slycot/tests/test_sb10yd.py:6]: See #207

**Comment by bnavigator** [comment]
Please rebase your feature branch onto master, there are some duplicate commits now

**Comment by KybernetikJo** [comment]
Yup, I have done something wrong.
I will try to fix it.

**Comment by bnavigator** [comment]
I fixed some linting warning in the tests, mainly in order to trigger a new CI run with a new commit. Somehow 6ff1902 was not picked up by GitHub Actions.

**Review by bnavigator — APPROVED** [review]
LGTM

**Pull request merged**